enterprise/server/util/redisutil/redisutil.go

```diff
@@ -68,43 +69,46 @@ func TargetToOptions(redisTarget string) *redis.Options {
 	}
 }
 
-func TargetToOpts(redisTarget string, useTLS bool) *Opts {
+func TargetToOpts(redisTarget string) *Opts {
 	if redisTarget == "" {
 		return nil
 	}
 	libOpts := TargetToOptions(redisTarget)
-	var tlsConfig *tls.Config
-	if useTLS {
-		tlsConfig = &tls.Config{
-			MinVersion: tls.VersionTLS12,
-		}
-	}
 	return &Opts{
 		Addrs:     []string{libOpts.Addr},
 		Network:   libOpts.Network,
 		Username:  libOpts.Username,
 		Password:  libOpts.Password,
 		DB:        libOpts.DB,
-		TLSConfig: tlsConfig,
+		TLSConfig: libOpts.TLSConfig,
 	}
 }
 
-func ShardsToOpts(shards []string, useTLS bool, username, password string) *Opts {
+func ShardsToOpts(shards []string, username, password string) *Opts {
 	if len(shards) == 0 {
 		return nil
 	}
-	var tlsConfig *tls.Config
-	if useTLS {
-		tlsConfig = &tls.Config{
-			MinVersion: tls.VersionTLS12,
+	var prevShardScheme string
+	for i, shard := range shards {
+		url, err := url.Parse(shard)
+		if err != nil {
+			log.Errorf("Failed to parse redis shard %q: %s", shard, err)
+			return nil
 		}
+
+		if i > 0 {
+			if url.Scheme != prevShardScheme {
+				log.Errorf("All redis shards must use the same url scheme, but found %q and %q", prevShardScheme, url.Scheme)
+				return nil
+			}
+		}
+		prevShardScheme = url.Scheme
 	}
-	return &Opts{
-		Addrs:     shards,
-		Username:  username,
-		Password:  password,
-		TLSConfig: tlsConfig,
-	}
+	opt := TargetToOpts(shards[0])
+	opt.Addrs = shards
```

Comment:
do we need to strip the scheme part of the URL for this `Addrs` list, or does the consuming function support schemes being present? asking because we aren't currently specifying schemes in our dev or prod configs.

Reply:
Good call.

I think it's best to add the stripping logic into `toRingOpts()` and use the official `redis.ParseURL` to better handle unix socket setups. Let me know if you have objection